kernel/os/include/os/arch/pic32/os/os_arch.h

```diff
@@ -59,7 +59,7 @@ typedef uint32_t os_stack_t;
 /* Exit a critical section, restore processor state and unblock interrupts */
 #define OS_EXIT_CRITICAL(__os_sr) __builtin_set_isr_state(__os_sr)
 /*  This is not the only way interrupts can be disabled */
-#define OS_IS_CRITICAL() ((__builtin_get_isr_state() & 1) == 0)
+#define OS_IS_CRITICAL() ((_CP0_GET_STATUS() & 1) == 0)
```

Comment:
It does this:

mfc0   $3, $12, 0
srl    $2,$3,10
ins    $2,$3,3,1
andi   $2,$2,0xf
sw     $2,0($fp)

At least 3 out of those 5 instructions are redundant, not a huge thing I know, but IMO its also clearer using _CP0_GET_STATUS(). Yes I can put it in a separate commit if you like.